From aeda0c69df2a6154e4f059dbea68aa8622794a9f Mon Sep 17 00:00:00 2001
From: Jan Wyszynski <jan.wyszynski@joyent.com>
Date: Wed, 21 Nov 2018 23:56:40 +0000
Subject: [PATCH] MANTA-4031 remove unnecessary muskie assert from disabled
 snaplinks check MANTA-4032 muskie_deleted_bytes metric should not include
 failed delete requests

---
 lib/audit.js  | 2 +-
 lib/common.js | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/audit.js b/lib/audit.js
index fcfee03..8a7f1bb 100644
--- a/lib/audit.js
+++ b/lib/audit.js
@@ -299,7 +299,7 @@ function auditLogger(options) {
             latency_histogram.observe(obj.latencyToFirstByte, labels);
         }
 
-        if (op === 'DELETE' && req.metadata) {
+        if (op === 'DELETE' && req.metadata && res.statusCode == 204) {
             var md = req.metadata;
             var owner = md.creator || md.owner;
 
diff --git a/lib/common.js b/lib/common.js
index 5df530b..b65e6f6 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -337,7 +337,6 @@ function checkAccountSnaplinksEnabled(req, uuid, next) {
     for (var i = 0; i < req.accountsSnaplinksDisabled.length; i++) {
         var account = req.accountsSnaplinksDisabled[i];
         assert.string(account.uuid, 'account.uuid');
-        assert.string(req.caller.account.uuid, 'req.caller.account.uuid');
 
         if (account.uuid === uuid) {
             next(false);
-- 
2.21.0

